src/main/scala/sigmastate/eval/CostingRules.scala

```diff
@@ -547,34 +557,38 @@ trait CostingRules extends SigmaLibrary { IR: RuntimeCosting =>
           })
           val values = obj.value.flatMap(f)
           val costs = colBuilder.replicate(sizes.length, 0)
-          RCCostedColl(values, costs, sizes, opCost(costOfArgs, costOf(method)))
+          RCCostedColl(values, costs, sizes, opCost(obj.value, costOfArgs, costOf(method)))
         case _ =>
```

Comment:
@greenhat values should be used in opCost

Reply:
Thanks for spotting this one as well! Another incorrect `assert` in `special` is fixed.